src/executor/mod.rs

```diff
@@ -0,0 +1,303 @@
+pub(crate) mod current_context;
+pub(crate) mod custom_executor;
+#[cfg(feature = "tokio")]
+pub(crate) mod spawn_blocking;
+
+use std::sync::OnceLock;
+
+use current_context::CurrentContextExecutor;
+use custom_executor::{CustomExecutor, CustomExecutorSyncClosure};
+
+use crate::{global_sync_strategy_builder, Error, ExecutorStrategy, GlobalStrategy};
+
+pub(crate) trait ExecuteSync {
+    /// Accepts a sync function and processes it to completion.
+    async fn execute_sync<F, R>(&self, f: F) -> Result<R, Error>
+    where
+        F: FnOnce() -> R + Send + 'static,
+        R: Send + 'static;
+}
+
+fn set_sync_strategy(strategy: SyncExecutor) -> Result<(), Error> {
```

Comment:
curious for your thoughts on naming — a `sync function` sounds like a function that synchronizes things. I wonder if we should say `synchronous`

Reply:
Sure, i'm fine with synchronous